lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java

```diff
@@ -165,16 +171,23 @@ private void launch(final File driverFolder,
       final LocalResource jarFileOnDFS = jobFolderOnDFS.uploadAsLocalResource(jarFile);
       LOG.info("Uploaded job submission JAR");
 
+      final Injector jobParamsInjector  = Tang.Factory.getTang().newInjector(jobSubmissionConfiguration);
 
       // ------------------------------------------------------------------------
       // Submit
-      submissionHelper
-          .addLocalResource(this.fileNames.getREEFFolderName(), jarFileOnDFS)
-          .setApplicationName(jobId)
-          .setDriverMemory(driverMemory)
-          .setPriority(priority)
-          .setQueue(queue)
-          .submit(ClientRemoteIdentifier.NONE);
+      try {
+        submissionHelper
+            .addLocalResource(this.fileNames.getREEFFolderName(), jarFileOnDFS)
+            .setApplicationName(jobId)
+            .setDriverMemory(driverMemory)
+            .setPriority(priority)
+            .setQueue(queue)
+            .setMaxApplicationAttempts(jobParamsInjector.getNamedInstance(MaxApplicationSubmissions.class))
```

Comment:
The call from C# to Java is getting unwieldy, yes. However, it shouldn't be too bad, as this `main()` is only ever called from C#, so we can evolve the command line parameters in lock-step.